src/Warpstone/Internal/ParserImplementations/WhitespaceParsersImpl.cs

```diff
@@ -0,0 +1,60 @@
+using System.Collections.Generic;
+using Warpstone.Errors;
+using Warpstone.Internal.ParserExpressions;
+
+namespace Warpstone.Internal.ParserImplementations
+{
+    /// <summary>
+    /// Represents a parser that parses whitespace.
+    /// </summary>
+    internal sealed class WhitespaceParsersImpl : ParserImplementationBase<WhitespacesParser, string>
+    {
+        /// <summary>
+        /// Singleton instance of the parser.
+        /// </summary>
+        public static readonly WhitespaceParsersImpl Instance = new();
```

Comment:
Add private default constructor to prevent other instances being created besides this singleton.